WEB/Src/DependencyCollector/DependencyCollector/Implementation/AzureSdk/AzureSdkDiagnosticsEventHandler.cs

```diff
@@ -325,6 +333,49 @@ private static bool IsMessagingDependency(string dependencyType)
                          dependencyType.EndsWith(RemoteDependencyConstants.AzureServiceBus, StringComparison.Ordinal));
         }
 
+        private static void SetCosmosDbProperties(Activity activity, DependencyTelemetry telemetry)
+        {
+            string dbAccount = null;
+            string dbName = null;
+            string dbOperation = null;
+            string dbContainer = null;
+
+            foreach (var tag in activity.Tags)
+            {
+                if (tag.Key == "db.name")
+                {
+                    dbName = tag.Value;
+                }
+                else if (tag.Key == "db.operation")
+                {
+                    dbOperation = tag.Value;
+                }
+                else if (tag.Key == "net.peer.name")
+                {
+                    dbAccount = tag.Value;
+                }
+                else if (tag.Key == "db.cosmosdb.container")
+                {
+                    dbContainer = tag.Value;
+                }
+                else if (tag.Key == "db.cosmosdb.status_code")
+                {
+                    telemetry.ResultCode = tag.Value;
+                    continue;
+                }
+                else if (!tag.Key.StartsWith("db.cosmosdb.", StringComparison.Ordinal))
+                {
+                    continue;
+                }
+                
+                telemetry.Properties[tag.Key] = tag.Value;
```

Comment:
if we are already populating db.name, and other things into target/name fields, do we need to store them again into property bag as well?

Reply:
Since `Target` and `Name` contain concatenated values optimized for portal UX and specific for AppInsights, we want to keep the original attributes as is for custom queries - e.g. we want users to be able to query by operation across containers or across all databases in account. This way Cosmos can document their telemetry once and it'd work for any backend: Azure Monitor, Jaeger, or Splunk.